packages/core/src/v4/entity.ts

```diff
@@ -0,0 +1,307 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { MapType } from '@sap-cloud-sdk/util';
+import { nonEnumerable } from '../util';
+import { toPropertyFormat } from '../util/name-converter';
+import { Constructable } from './constructable';
+import { EntityBuilder } from './entity-builder';
+import { CustomField, Link, Field } from './selectable';
+
+export type EntityBuilderType<
+  EntityT extends Entity,
+  EntityTypeForceMandatoryT
+> = {
+  [property in keyof EntityTypeForceMandatoryT]: (
+    value: EntityTypeForceMandatoryT[property]
+  ) => EntityBuilderType<EntityT, EntityTypeForceMandatoryT>;
+} &
+  EntityBuilder<EntityT, EntityTypeForceMandatoryT>;
+
+/**
+ * Super class for all representations of OData entity types.
+ */
+export abstract class Entity {
+  static _serviceName: string;
```

Comment:
Perhaps you can add an interface or parent class which holds the common fields of the Entity in v2 and v4 so that one has less duplication